Formula/fnm.rb

```diff
@@ -0,0 +1,53 @@
+class Fnm < Formula
+  desc "Fast and simple Node.js version manager"
+  homepage "https://fnm.vercel.app"
+  url "https://github.com/Schniz/fnm/archive/v1.22.2.tar.gz"
+  sha256 "49013376359a1e0d690359b10c09604df90ab30602e8400acf0a97d241707344"
+  license "GPL-3.0-only"
+  head "https://github.com/Schniz/fnm.git"
+
+  livecheck do
+    url "https://github.com/Schniz/fnm/releases/latest"
+    regex(%r{href=.*?/tag/v?(\d+(?:\.\d+)+)["' >]}i)
+  end
+
+  depends_on "rust" => :build
+
+  uses_from_macos "zlib"
+
+  def install
+    system "cargo", "install", *std_cargo_args
+
+    (bash_completion/"fnm").write `#{bin}/fnm completions --shell bash`
+    (fish_completion/"fnm.fish").write `#{bin}/fnm completions --shell fish`
+    (zsh_completion/"_fnm").write `#{bin}/fnm completions --shell zsh`
+  end
+
+  def caveats
+    <<~CAVEATS
+      Thanks for installing fnm!
+      The last step for making fnm work is to run it on the shell startup, using the `fnm env` command.
+
+      In order to complete the installation, please add the following to your shell profile:
+
+        # fnm
+        #{source_for_shell}
+
+      Homebrew tells us that #{preferred} is your preferred shell,
+      and your shell profile is at #{shell_profile.inspect}, if that helps 😇
+    CAVEATS
+  end
```

Comment:
This isn't homebrew specific so it shouldn't be a caveat.

Reply:
But it is needed to finish the installation. Is there anything else I can do to let the users know that they should do it? This is based on: https://github.com/Homebrew/homebrew-core/blob/master/Formula/nvm.rb#L16-L39

Reply:
This should be in the upstream documentation.

Reply:
@SMillerDev can the installation via Homebrew at least mention the need for the shell profile integration, with a link to the upstream docs explaining how to do it?

Reply:
No, unless there is a non-standard situation caused by Homebrew as described in https://docs.brew.sh/Formula-Cookbook#caveats

Reply:
Okay.
I pushed a change. Thanks!